### PR TITLE
Multithreaded Pokemon searching

### PR DESCRIPTION
**Pull request by hgentry**
This pull request includes a

- [ ] Bug fix
- [x] New feature
- [ ] Translation

The following changes were made

I've added support for multithreaded requests as suggested in #357, which greatly speeds up the process. I have 15-step maps running in the time a 6-step map took before multithreading. To run with multithreading, add -mt to your arguments.

### CODE REVIEW

File: example.py [hunk 1]
```diff
@@ -471,6 +471,12 @@ def get_args():
         help='Set web server listening host',
         default='127.0.0.1')
     parser.add_argument(
+        '-mt',
+        '--multithreading',
+        help='Enable multithreaded search',
+        action='store_true',
+        default=False)
```
Comment (juharris) [inline example.py]: Why not default to `True`?

Comment (hgentry) [inline example.py]: Just in case there's a major stability issue I didn't catch. I don't want to break everyone's maps until this has been used more.

**Comment by medghaim** [comment]
This is awesome! But unfortunately the entire project is being rewritten, example.py won't exist soon. Check out the rewrite-branch, as per the readme.

**Comment by hgentry** [comment]
I look forward to contributing to the new project! Is there an ETA?

**Pull request closed without merging**